cmd/zscrub/zscrub.c

```diff
@@ -0,0 +1,242 @@
+    int nsearchdirs = 0;
+    char *poolname = NULL;
+    int do_restart = 0;
+
+    spa_t *spa = NULL;
+
+    zfs_scrub_ddt_class_max = DDT_CLASS_UNIQUE; // Scan the entire DDT
+    reference_tracking_enable = FALSE; // Disable reference tracking debugging
+    zfs_no_scrub_prefetch = B_TRUE; // Disable prefetch during scan
+
+    {
+        char c;
+        while ((c = getopt(argc, argv, "D:PTp:rv")) != -1) {
+            switch(c) {
+            case 'D':
+                // How much of the DDT are we scanning?
+                {
+                    char *endptr = NULL;
+                    unsigned long class = strtoul(optarg, &endptr, 0);
+                    if ((errno == 0)
+                       && (*endptr == '\0')
+                       && (class < DDT_CLASSES)) {
+                        zfs_scrub_ddt_class_max = class;
+                    } else {
+                        (void) fprintf(stderr,
+                            "DDT class must be between 0 and %d inclusive\n",
+                            DDT_CLASSES-1);
+                        exit(1);
+                    }
+                }
+                break;
+            case 'P':
+                zfs_no_scrub_prefetch = B_FALSE; // Turn prefetching back on
+                break;
+            case 'T':
+                reference_tracking_enable = FALSE; // Turn reference tracking back on
+                break;
+            case 'p':
+                // Add a search path
+                {
+                    char **tmp = realloc(searchdirs,
+                                         (nsearchdirs + 1) * sizeof (char *));
+                    if (tmp == NULL) {
+                        (void) fprintf(stderr, "Out of memory\n");
+                        exit(1);
+                    }
+                    searchdirs = tmp;
+                    searchdirs[nsearchdirs++] = optarg;
+                }
+                break;
+            case 'r':
+                // Restart a scrub
+                do_restart++;
+                break;
+            case 'v':
+                // Be chatty
+                verbose++;
+                break;
+            case '?':
+                (void) fprintf(stderr, "invalid option '%c'\n", optopt);
+                exit(1);
+            }
+        }
+    }
+
+    if (optind == argc) {
+        (void) fprintf(stderr, "Need pool name\n");
+        exit(1);
+    }
+    if (optind + 1 < argc) {
+        (void) fprintf(stderr, "WARNING: Discarding excess arguments\n");
+    }
+    poolname = argv[optind];
+
+    if (verbose && (searchdirs != NULL)) {
+        int sdix = 0;
+        fprintf(stderr, "Will search:\n");
+        for (sdix = 0; sdix < nsearchdirs; sdix++) {
+            fprintf(stderr, "\t%s\n", searchdirs[sdix]);
+        }
+    }
+
+    dprintf_setup(&argc, argv);
+    kernel_init(FREAD | FWRITE | FCREAT);
+    VERIFY(libzfs_handle = libzfs_init());
+
+    {
+      importargs_t ia;
+      memset(&ia, 0, sizeof(ia));
```

Comment:
Purely personal preference, but I prefer the `= {0}` static initializer to bzero or memset.